tests/Integration/Tagging/TagDocumentTest.php

```diff
@@ -0,0 +1,228 @@
+<?php declare(strict_types=1);
+
+namespace Integration\Tagging;
+
+use Neusta\Pimcore\HttpCacheBundle\CacheActivator;
+use Neusta\Pimcore\HttpCacheBundle\Element\Document\DocumentType;
+use Neusta\Pimcore\TestingFramework\Database\ResetDatabase;
+use Neusta\Pimcore\TestingFramework\Test\Attribute\ConfigureExtension;
+use Neusta\Pimcore\TestingFramework\Test\Attribute\ConfigureRoute;
+use Neusta\Pimcore\TestingFramework\Test\ConfigurableWebTestcase;
+use Pimcore\Cache\RuntimeCache;
+use Pimcore\Model\Document;
+
+final class TagDocumentTest extends ConfigurableWebTestcase
+{
+    use ResetDatabase;
+
+    /**
+     * @test
+     *
+     * @dataProvider documentsTypeProvider
+     */
+    #[ConfigureRoute(__DIR__ . '/Fixtures/get_document_route.yaml')]
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'assets' => false,
+            'documents' => true,
+            'objects' => false,
+        ],
+    ])]
+    public function response_is_tagged_with_expected_tags_when_document_is_loaded(DocumentType $type): void
+    {
+        $client = self::createClient();
+
+        $document = new Document();
+        $document->setId(42);
+        $document->setKey('test_document');
+        $document->setPublished(true);
+        $document->setParentId(1);
+        $document->setType($type->value);
```

Comment:
I don't think you can change the type of document this way 😕 

I think you have to create the appropriate objects via `new` instead.